File: packages/php/src/Exception/Origin.php
<?php

declare(strict_types=1);

namespace Xyp\Exception;

/**
 * Whose side a problem is on, so logs and dashboards can route it.
 */
enum Origin: string
{
    /** How the client was set up: a bad key, an unknown operation, a bad parameter. */
    case Config = 'config';
    /** The VPN, the hosts entry, DNS, TLS or a timeout. */
    case Network = 'network';
    /** XYP or the data provider answering with an error. */
    case Xyp = 'xyp';
    /** A gap in this SDK. Please report it. */
    case Sdk = 'sdk';
}

File: packages/java/src/main/java/io/github/baljinnyamday/xyp/Origin.java
package io.github.baljinnyamday.xyp;

/**
 * Whose side a problem is on, so logs and dashboards can route it. Every {@link XypException}
 * carries one; {@link XypException#originOf(Throwable)} works on any throwable.
 */
public enum Origin {
  /** How the client or the call was set up: a bad key, an unknown operation, a bad parameter. */
  CONFIG,
  /** The VPN, the hosts entry, DNS, TLS or a timeout: XYP could not be reached. */
  NETWORK,
  /** XYP or the data provider answering with an error. */
  XYP,
  /** A gap in this SDK. Please report it. */
  SDK
}
